src/main/kotlin/com/demonwav/mcdev/i18n/I18nElementFactory.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.i18n
+
+import com.demonwav.mcdev.i18n.lang.I18nFile
+import com.demonwav.mcdev.i18n.lang.I18nFileType
+import com.demonwav.mcdev.i18n.lang.gen.psi.I18nProperty
+import com.demonwav.mcdev.util.runWriteAction
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.popup.JBPopupFactory
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFileFactory
+import com.intellij.psi.PsiManager
+import com.intellij.psi.search.FileTypeIndex
+import com.intellij.psi.search.GlobalSearchScope
+import com.intellij.util.indexing.FileBasedIndex
+import javax.swing.JList
+
+object I18nElementFactory {
+    val DOMAIN_PATTERN = Regex("^.*?/assets/(.*?)/lang.*?\$")
+
+    fun getResourceDomain(file: VirtualFile) =
+        DOMAIN_PATTERN.matchEntire(file.path)?.groupValues?.get(1)
+
+    fun addTranslation(module: Module?, name: String, value: String?) {
+        if (module == null || value == null)
```

Comment:
Please never omit the braces on if statements (with the exception of ternary style statements).